src/structures/Role.js

```diff
@@ -197,8 +197,8 @@ class Role extends Base {
    *   .catch(console.error);
    */
   async edit(data, reason) {
-    if (typeof data.permissions !== 'undefined') data.permissions = Permissions.resolve(data.permissions);
-    else data.permissions = this.permissions.bitfield;
+    if (typeof data.permissions !== 'undefined') data.permissions = new Permissions(data.permissions);
+    else data.permissions = this.permissions;
```

Comment:
Might as well remove the `if`/`else` and pass this directly within the `data` object [below](https://github.com/vaporox/discord.js/blob/fix-edit/src/structures/Role.js#L223) (this would also avoid any unwanted mutation)
```diff
- permissions: data.permissions,
+ permissions: typeof data.permissions !== 'undefined' ? new Permissions(data.permissions) : this.permissions
```